tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -36,20 +50,59 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
         else:
             resource_name = self.moniker
 
-        with self.override_disable_recording():
-            retries = 4
-            for i in range(retries):
-                try:
-                    parameter_update = self.create_resource(
-                        resource_name,
-                        **kwargs
-                    )
-                    break
-                except AzureNameError:
-                    if i == retries - 1:
+        # If a child is cached we must use the same cached resource their equivalent parent did so all the deps line up
+        try:
```

Comment:
I would prefer to see this block in "_preparer_wrapper" and not add "fn" as a parameter of "_prepare_create_resource". I actually split "_preparer_wrapper" into "_preparer_wrapper/_prepare_create_resource" not that long ago because I wanted to have some preparation code available without having to fit "fn"